### ISSUE TITLE
NoClassDefFoundError: FilterRegistrationBean

### ISSUE THREAD

**Issue by FdezI** [grails-plugins/grails-spring-security-core#528]
I'm experimenting a weird issue with (Grails version is: 3.3.2):

```
    compile 'org.grails.plugins:spring-security-core:3.2.0'
    compile('org.grails.plugins:spring-security-shiro:3.0.1') {
        exclude group: 'org.grails.plugins', module:'spring-security-core'
    }
```

The trace is:

```
:bootRun

Configuring Spring Security Core ...
2018-01-22 14:46:59.024 ERROR --- [           main] o.s.boot.SpringApplication               : Application startup failed

java.lang.NoClassDefFoundError: org/springframework/boot/context/embedded/FilterRegistrationBean
	at grails.plugin.springsecurity.SpringSecurityCoreGrailsPlugin$_doWithSpring_closure12.doCall(SpringSecurityCoreGrailsPlugin.groovy:204)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springsource.loaded.ri.ReflectiveInterceptor.jlrMethodInvoke(ReflectiveInterceptor.java:1427)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
	at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:294)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1022)
	at groovy.lang.Closure.call(Closure.java:414)
	at groovy.lang.Closure.call(Closure.java:408)
	at grails.spring.BeanBuilder.invokeBeanDefiningClosure(BeanBuilder.java:759)
	at grails.spring.BeanBuilder.beans(BeanBuilder.java:588)
	at grails.spring.BeanBuilder.invokeMethod(BeanBuilder.java:531)
	at org.grails.plugins.DefaultGrailsPlugin.doWithRuntimeConfiguration(DefaultGrailsPlugin.java:559)
	at org.grails.plugins.AbstractGrailsPluginManager.doRuntimeConfiguration(AbstractGrailsPluginManager.java:167)
	at grails.boot.config.GrailsApplicationPostProcessor.postProcessBeanDefinitionRegistry(GrailsApplicationPostProcessor.groovy:171)
	at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanDefinitionRegistryPostProcessors(PostProcessorRegistrationDelegate.java:272)
	at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanFactoryPostProcessors(PostProcessorRegistrationDelegate.java:122)
	at org.springframework.context.support.AbstractApplicationContext.invokeBeanFactoryPostProcessors(AbstractApplicationContext.java:687)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:525)
	at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.refresh(EmbeddedWebApplicationContext.java:122)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:693)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:360)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:303)
	at grails.boot.GrailsApp.run(GrailsApp.groovy:83)
	at grails.boot.GrailsApp.run(GrailsApp.groovy:387)
	at grails.boot.GrailsApp.run(GrailsApp.groovy:374)
	at grails.boot.GrailsApp$run.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:133)
	at com.code.Application.main(Application.groovy:8)
Caused by: java.lang.ClassNotFoundException: org.springframework.boot.context.embedded.FilterRegistrationBean
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:335)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 34 common frames omitted
```

If I follow the trace and go to: `SpringSecurityCoreGrailsPlugin.groovy:204`, I find that the class being imported is:

`import org.springframework.boot.web.servlet.FilterRegistrationBean`

However, somehow, the runtime expects the non-existent: 

`org.springframework.boot.context.embedded.FilterRegistrationBean`

**Comment by puneetbehl** [comment]
@FdezI can you please verify the problem using Grails v3.3.2 and spring-security-core plugin version to `3.2.1.BUILD-SNAPSHOT`. Also, it would be great if you could share the sample application replicating the problem.

**Comment by sdelamo** [comment]
which Grails version are you using?

**Comment by sdelamo** [comment]
I think the issue maybe the shiro plugin which is not updated to 3.3. 

We experienced this issue in Spring Security Core: 
https://github.com/grails-plugins/grails-spring-security-core/issues/488

**Comment by FdezI** [comment]
I'm using Grails 3.3.2 (as stated in the OP). However, I think we can close this issue, I'm not using shiro any more.

**Comment by puneetbehl** [comment]
This problem has been fixed in Grails Spring Security Shiro v3.1.1
